run.go

```diff
@@ -0,0 +1,194 @@
+package pack
+
+import (
+	"context"
+	"crypto/md5"
+	"fmt"
+	"io"
+	"log"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/buildpack/pack/config"
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/go-connections/nat"
+	"github.com/pkg/errors"
+)
+
+type RunFlags struct {
+	AppDir   string
+	Builder  string
+	RunImage string
+	Port     string
+}
+
+type RunConfig struct {
+	Port  string
+	Build Task
+	// All below are from BuildConfig
+	AppDir          string
+	Builder         string
+	RunImage        string
+	RepoName        string
+	Cli             Docker
+	Stdout          io.Writer
+	Stderr          io.Writer
+	Log             *log.Logger
+	FS              FS
+	Config          *config.Config
+	Images          Images
+	WorkspaceVolume string
+	CacheVolume     string
+}
+
+func (bf *BuildFactory) RunConfigFromFlags(f *RunFlags) (*RunConfig, error) {
+	bc, err := bf.BuildConfigFromFlags(&BuildFlags{
+		AppDir:   f.AppDir,
+		Builder:  f.Builder,
+		RunImage: f.RunImage,
+		RepoName: f.repoName(),
+		Publish:  false,
+		NoPull:   false,
+	})
+	if err != nil {
+		return nil, err
+	}
+	rc := &RunConfig{
+		Build: bc,
+		Port:  f.Port,
+		// All below are from BuildConfig
+		AppDir:          bc.AppDir,
+		Builder:         bc.Builder,
+		RunImage:        bc.RunImage,
+		RepoName:        bc.RepoName,
+		Cli:             bc.Cli,
+		Stdout:          bc.Stdout,
+		Stderr:          bc.Stderr,
+		Log:             bc.Log,
+		FS:              bc.FS,
+		Config:          bc.Config,
+		Images:          bc.Images,
+		WorkspaceVolume: bc.WorkspaceVolume,
+		CacheVolume:     bc.CacheVolume,
+	}
```

Comment:
I not sure why some of these flags are needed on RunConfig. It seems like `AppDir`, `Builder`, `RunImage`,  `Config`, `Images`, `WorkspaceVolume` and `CacheVolume` aren't needed by the implementation of Run. I think they are only needed to generate the BuildConfig, which you are already saving on RunConfig. Let me know if I am missing something...

Reply:
You're right, they are not used. I'll remove them.